nodes/orchestrator.py

```diff
@@ -56,6 +57,29 @@ def orchestrate(
 
     # If human action is required, run the actions summarizer and prompt the human to take action
     if extended_category.human_action_required:
+        # If actions summary exists but referral possibility hasn't been assessed yet
+        if state.actions_summary is not None and state.referral_possibility is None:
+            # Run referral possibility inference after actions were taken
+            state.referral_possibility = infer_referral_possibility(
+                state.context, 
+                state.classified_category, 
+                state.actions_summary, 
+                dry_run=False
+            )
+            state.step = "next: referral possibility assessed"
+            return state
+        
+        # If referral possibility has been assessed, suggest topics based on the result
+        if state.referral_possibility is not None:
+            state.suggested_topics = suggest_topics(
+                state.context, 
+                state.classified_category, 
+                state.referral_possibility,
+                dry_run=False
+            )
+            state.step = "next: select topics"
+            return state
+        
```

Comment:
Should we first check if the inference result exists, like the following diagram?

<img width="1006" height="379" alt="Screenshot 2025-08-20 at 10 54 50 AM" src="https://github.com/user-attachments/assets/0f040ff3-c8a5-4238-87dd-455985cdce52" />